connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorPluginsResource.java

```diff
@@ -50,7 +50,7 @@ public ConnectorPluginsResource(Herder herder) {
     public ConfigInfos validateConfigs(final @PathParam("connectorType") String connType,
                                        final Map<String, String> connectorConfig) throws Throwable {
         String includedConnType = connectorConfig.get(ConnectorConfig.CONNECTOR_CLASS_CONFIG);
-        if (includedConnType != null && !includedConnType.equals(connType))
+        if (includedConnType != null && !includedConnType.endsWith(connType))
```

Comment:
Given that we have both `ConnectorFactory` and `PluginDiscovery` which should be able to give you the full set of both canonical and short names, couldn't we just do a look up both on `connType` and `includedConnType` and validate that they are the same? It might require a bit of reworking (e.g. making `Map<String, Class<? extends Connector>>` available instead of `List<Class<? extends Connector>>` in PluginDiscovery), but it seems like a better solution since it'll address all the cases afaict.

re: BadRequestException, we needed something in the 400 range and nothing else seemed appropriate.